[SYCL][Doc] Announce ABI-breaking window (#6274)
intel/llvm: sycl/doc/developer/ABIPolicyGuide.md

  Please, prefer updating the test files with the above command. The checker
  script automatically sorts symbols. This would allow developers to avoid
  large diffs and help maintainers identify the nature of ABI changes.
* `test/abi/layout*` and `test/abi/symbol_size_alignment.cpp` are a group of
  tests to check the internal layout of some classes. The `layout*` tests check 
  some of API classes for layout changes, while `symbol_size_alignment` only
  checks `sizeof` and `alignof` for API classes. Changing the class layout is a 
  breaking change.

## Changing ABI

Generally DPC++ runtime and compiler ABI is frozen and ABI-breaking changes are

> not accepted by default since Oct 2020. Please, try to avoid making any breaking

Reviewer: ```suggestion
not accepted by default since Oct 2020. Please try to avoid making any breaking
```
Not completely sure I am right here but the comma looks off. 😄

changes. If you need to change existing functionality, consider adding new APIs

> instead of replacing them. Also, please, avoid any changes, mentioned in the

Reviewer: ```suggestion
instead of replacing existing APIs. Also, please, avoid any changes, mentioned in the
```

Reviewer: There's also something about the commas on the following sentence. @cperkinsintel - How strong are your English comma rules?

[Intro](#intro) section as breaking. Refer to the above guide to distinguish

> breaking and non-breaking changes. If not sure, do not hesitate to ask code

Reviewer: ```suggestion
between breaking and non-breaking changes. If unsure, do not hesitate to ask code
```

owners for help.

**Note**: Features clearly marked as experimental are considered as an exception
to this guideline.

### ABI breaking changes window June 15 - September 15 2022

> Since the point when ABI was frozen ~2 years ago we accumulated significant

Reviewer: ```suggestion
Since the point when ABI was frozen ~2 years ago we have accumulated a significant
```

volume of changes, which require breaking ABI. We are opening a window to
implement these changes. During this period of time ABI breaking changes will be

> accepted. All breaking changes still require a good justification.

Reviewer: ```suggestion
accepted. All breaking changes will still require a good justification.
```
